lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -370,13 +371,15 @@
       - Rolling update parallelism.
       - Corresponds to the C(--update-parallelism) option of C(docker service create).
   update_failure_action:
-    type: int
+    type: str
+    default: pause
     description:
       - Action to take in case of container failure.
       - Corresponds to the C(--update-failure-action) option of C(docker service create).
     choices:
       - continue
       - pause
+      - rollback
```

Comment:
This needs to be added to the `update_failure_action ` choices in `argument_spec` as well.

Usage of the `rollback` choice also needs to be checked in `option_minimal_versions`. See how `publish_mode` is implemented.